src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,167 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int listLength;
+
     @Override
     public void add(T value) {
-
+        if (listLength < array.length) {
+            array[listLength] = value;
+            listLength++;
+        } else {
+            array = grow(array);
+            array[listLength] = value;
+            listLength++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + listLength);
+        }
+        if (index < array.length && index < listLength) {
+            if (listLength == array.length) {
+                array = grow(array);
+            }
+            for (int i = array.length - 1; i > index; i--) {
+                array[i] = array[i - 1];
+            }
+            array[index] = value;
+            listLength++;
+        } else if (index == listLength) {
+            this.add(value);
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + listLength);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (list.size() + listLength > array.length) {
+            array = grow(array);
+        }
+        if (listLength == 0) {
+            for (int i = 0; i < list.size(); i++) {
+                array[i] = list.get(i);
+                listLength++;
+            }
+        } else {
+            int shift = indexOf(null);
+            for (int i = 0; i < list.size(); i++) {
+                array[i + shift] = list.get(i);
+            }
+            listLength += list.size();
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        T value = null;
+        int selectedIndex = -1;
+        for (int i = 0; i < this.size(); i++) {
+            if (i == index) {
+                value = array[i];
+                selectedIndex = i;
+            }
+        }
+        if (selectedIndex == -1) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + listLength);
+        }
+        return value;
     }
 
     @Override
     public void set(T value, int index) {
-
+        int updatedIndex = -1;
+        for (int i = 0; i < listLength; i++) {
+            if (i == index) {
+                array[i] = value;
+                updatedIndex = i;
+            }
+        }
+        if (updatedIndex == -1) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + listLength);
+        }
```

Comment:
Similar to the 'get' method, the 'set' method's logic can be simplified. You should check if the index is within bounds at the start and directly set the value if it is valid, throwing an exception otherwise. The 'updatedIndex' variable is unnecessary.